design/adr/core/0002-Developer-Scripts-TaggingAndCleanup.md

```diff
@@ -0,0 +1,105 @@
+
+## Proposed Design 
+
+### Force using annotated tags
+
+By using `annotated tags` git history will be marked with the time versions have been released. Exp: git tag -a v2.0 -m "Releasing version v2.0" <br />
+Developers will have the knowledge which specific commit to use for specific EdgeX version. 
+This will make the troubleshooting possible for both - the older and the current versions of the project.
+
+### Clearing Up and Update the scripts
+Some of the scripts need to be updated in order to work with the latest version, other are not relevant at all and need to be removed.
+
+##### Current available scripts 
+```
+MODULES*
+clean_shell.bat
+mongoshell.bat
+init_shell.bat
+startdb-no-auth.bat
+startdb.bat
+clean_mongo.js
+init_mongo.js
+linux_setup.sh*
+mongoAdminOperStateUpdate.js
+run-it.sh*
+reset-dockers.sh*
+prepare-environment.sh*
+create-containers.sh*
+update-packages.sh*
+```
+
+##### Proposed changes per script 
+* [MODULES](https://github.com/edgexfoundry/developer-scripts/blob/master/MODULES) - **to be Updated** <br /> 
+Still contains java bases modules. These needs to be replace with the latest go applications
+
+* [mongoshell.bat](https://github.com/edgexfoundry/developer-scripts/blob/master/mongoshell.bat), 
+    [init_shell.bat](https://github.com/edgexfoundry/developer-scripts/blob/master/init_shell.bat),
+    [startdb-no-auth.bat](https://github.com/edgexfoundry/developer-scripts/blob/master/startdb-no-auth.bat),
+    [startdb.bat](https://github.com/edgexfoundry/developer-scripts/blob/master/startdb.bat),
+    [init_mongo.js](https://github.com/edgexfoundry/developer-scripts/blob/master/init_mongo.js),
+    [linux_setup.sh](https://github.com/edgexfoundry/developer-scripts/blob/master/linux_setup.sh) - **to be Removed**. <br /> 
+ All these scripts set up mongo database. Instead, go application [docker-edgex-mongo](https://github.com/edgexfoundry/docker-edgex-mongo) is supposed to be used.
+* [clean_mongo.js](https://github.com/edgexfoundry/developer-scripts/blob/master/clean_mongo.js)  - **to be Removed** <br />
+These file should be removed once a second executable is added in [docker-edgex-mongo](https://github.com/edgexfoundry/docker-edgex-mongo). Currently `docker-edgex-mongo` is single purpose application responsible for initializing mongo database (setting up users/passwords/privileges and creates databases and collections). In addition, executable for `purging` the entire database need to be added. It is supposed to work in both security enabled/disabled environments.
+It could be used by Edgex CLI for purge the entire database (without the need to have all the microservices up and running). More about this proposal [here](0002-MongoDB-Clearance.md)  
+
+* mongoAdminOperStateUpdate.js - **to be Removed** <br />
+The script is not relevant anymore
+
+* [prepare-environment.sh](https://github.com/edgexfoundry/developer-scripts/blob/master/prepare-environment.sh) -  **to be Modified** <br />
```

Comment:
Can we change the name to something more applicable like `clone-modules`?

Reply:
agree

Reply:
but we could consider `clone-modules` as a little bit misleading name - this scripts clone or pulls the project if they already exists

Reply:
OK, so then `update-packages.sh` is redundant.  I propose renaming this one to `get-modules.sh` and remove `update-packages.sh`

Reply:
Agree